src/errors/RequestResponseError.ts

```diff
@@ -4,50 +4,81 @@
+};
+
+type FormattedErrorRequest = Omit<ErrorResponse, "status">;
+
+const isSymbol = (value: unknown): value is string | number | boolean =>
+    ["string", "number", "boolean"].includes(typeof value);
+
+/**
+ * The function format the ill-formatted errors:
+ * - When errors is not an array, use it as the only error
+ * - Filters null and undefined errors
+ * - When an item of errors is a symbol rather than an object, returns the UNKNOWN_ERROR as the code
+ */
+const formatErrors = ({ errors: rawErrors, ...rest }: RawErrorRequest): FormattedErrorRequest => {
+    const errorList = Array.isArray(rawErrors) ? rawErrors : [rawErrors];
+
+    const formattedErrors: (SubError | ValidationError)[] = errorList.filter(Boolean).map((error: ErrorItem) => {
+        if (typeof error === "string") {
+            const errorCodeKeys = [...Object.keys(ResponseErrorCode), ...Object.keys(RequestErrorCode)];
+            const isValidError = errorCodeKeys.some(
+                (key) => ResponseErrorCode[key] === error || RequestErrorCode[key] === error
+            );
+
+            return {
+                reason: isValidError ? (String(error) as ResponseErrorCode) : ResponseErrorCode.UnknownError,
+                field: null,
+            };
+        }
+
+        if (isSymbol(error)) {
+            return { reason: ResponseErrorCode.UnknownError, field: null };
         }
 
-        // (pH) FIXME: Remove this once the ts services will be fixed
-        const errorList = Array.isArray(errors) ? errors : [errors];
+        return error;
+    });
 
-        return errorList.filter(Boolean).map((error: SubError | ValidationError | symbol) => {
-            if (isSymbol(error)) {
-                return error; // (pH) FIXME: special handling for wrongly formatted. Fix the ErrorRequest type.
-            }
+    return { ...rest, errors: formattedErrors };
+};
 
-            return "field" in error ? `${error.reason} (${error.field})` : `${error.reason}`;
-        });
-    })();
+const serializeErrorResponse = ({ code, httpCode, errors }: FormattedErrorRequest): string => {
+    const formattedErrors = errors.map((error: SubError | ValidationError) =>
+        "field" in error && !!error.field ? `${error.reason} (${error.field})` : `${error.reason}`
+    );
 
     return `Code: ${code}, HttpCode: ${httpCode}, Errors: ${formattedErrors.join(", ")}`;
 };
 
 export class RequestResponseBaseError extends Error {
     errorResponse: ErrorResponse;
 
-    constructor(errorResponse: ErrorRequest) {
-        super(serializeErrorResponse(errorResponse));
+    constructor(errorResponse: RawErrorRequest) {
+        const formattedErrorResponse = formatErrors(errorResponse);
+        super(serializeErrorResponse(formattedErrorResponse));
+
         Object.setPrototypeOf(this, RequestResponseBaseError.prototype);
-        this.errorResponse = { status: "error", ...errorResponse };
+        this.errorResponse = { status: "error", ...formattedErrorResponse };
     }
 }
```

Comment:
Ok, wha changed here is that the error is formatted to fit the right type. Before the `serializeErrorResponse` was handling this but the error inside the `RequestResponseBaseError` still was ill-formatted that can cause issues in the consumers when using the object as the type was wrong.

Now it should be consistent with the information typescript is giving.

Reply:
Note that more tests have been added here as well.